### DISCUSSION TITLE
how to fix the error `importSource cannot be set when runtime is classic.`

### DISCUSSION THREAD

**Discussion by hyeongrok7874** [storybookjs/storybook#23434] [Help]
Previously, css prop of emotion was used by setting jsxImportSource in tsconfig.

However, for some reason, the option is not available, so `/** @jsxImportSource @emotion/react */` pragma is written at the top level of each file that uses css props.

After that, when running or building the storybook, the error `importSource cannot be set when runtime is classic.` occurs.

I would appreciate it if you could tell me how to solve the above error.

**Comment by noahbald** [comment]
Hey @hyeongrok7874 did you ever figure this out?

**Reply by hyeongrok7874** [reply]
@noahbald

Yes, it is currently resolved.

In my environment, the framework was set to '@storybook/nextjs' in the config of `.storybook/main.js`.

When relying on '@storybook/nextjs', we confirmed that the problem occurred because the `preset-react` property of `next/babel` was set to `runtime: 'classic'` in babel's default preset.

Therefore, the problem no longer occurs by overwriting the existing babel preset as shown below.

.storybook/main.js
```js
  babel: async (config) => ({
    ...config,
    presets: [
      [
        'next/babel',
        {
          'preset-react': {
            runtime: 'automatic',
          },
        },
      ],
    ],
  }),
```

I'm attaching my pull request that solved the issue in case it helps.

https://github.com/themoment-team/official-gsm-front/pull/185/files

**Reply by noahbald** [reply]
This didn't end up actually working for me, but I figured out that the base preset is a function that you can pass options to, which worked for me :)
```ts
async babel(config) {
	const basePreset = config.presets?.[0];
	return {
		...config,
		presets: [
			(api, options) => {
				const output =
					typeof basePreset === 'function' &&
					basePreset(api, {
						...options,
						'preset-react': { runtime: 'automatic' },
					});
				return output;
			},
		],
	};
}
```

**Reply by hyeongrok7874** [reply]
Thank you so much for sharing this great method!
